plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingsProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.trino.plugin.base.util.JsonUtils;
+
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+public abstract class S3SecurityMappingsProvider
```

Comment:
Can we use composition instead of inheritance here?

Reply:
How exactly would you like that to look? Remove `abstract` from `S3SecurityMappingsProvider` and create a new interface for `UriBasedS3SecurityMappingsProvider` and `FileBasedS3SecurityMappingsProvider` to implement?

Reply:
I think it is fine to have an interface for `S3SecurityMappingsProvider` that is only `implements Supplier<S3SecurityMappings>`, but then when implementing the other classes we could extract the common code to a separate class that is used from `UriBasedS3SecurityMappingsProvider` and `FileBasedS3SecurityMappingsProvider`

Reply:
Alright, I will update the PR accordingly